homeassistant/components/rainmachine/switch.py

```diff
@@ -123,6 +123,17 @@ class RainMachineSwitchDescription(
     """Describe a RainMachine switch."""
 
 
+_T = TypeVar("_T")
+
+
+@callback
+def async_limit_precision(value: _T) -> _T:
```

Comment:
FYI: I wrote this helper because I've seen instances where some of the attributes I'm modifying don't exist on certain devices.

Reply:
Does it make sense to not add those attributes to `_attr_extra_state_attributes` when they are not present ?

Reply:
The hard part is that they seem to exist for some controllers and not others. But, it's a fair point: I could be a little smarter about checking each and only including something if it exists.